### ISSUE TITLE
BibTeX "number" field should be mapped to CSL "issue" field?

### ISSUE THREAD

**Issue by hubgit** [dsifford/astrocite#28]
In BibTeX the `number` field contains "The number of a journal, magazine, technical report, or of a work in a series. An issue of a journal or magazine is usually identified by its volume and number; the organization that issues a technical report usually gives it a number; and sometimes books are given numbers in a named series."

https://www.openoffice.org/bibliographic/bibtex-defs.html

In CSL the `issue` field contains "(container) issue holding the item (e.g. “5” when citing a journal article from journal volume 2, issue 5)" and the `number` field contains "number identifying the item (e.g. a report number)". 

http://docs.citationstyles.org/en/stable/specification.html#appendix-iv-variables

It seems like BibTeX's `number` field maps best to CSL's `issue` field, but currently `astrocite-bibtex` maps `number` to `number`.

**Code**

```js
const { parse } = require('astrocite-bibtex')

const items = parse(`@article{test,
author = {Arch, Joanna J and Wolitzky-Taylor, Kate B and Eifert, Georg H and Craske, Michelle G},
title = {{Longitudinal treatment mediation of traditional cognitive behavioral therapy and acceptance and commitment therapy for anxiety disorders.}},
journal = {Behaviour research and therapy},
year = {2012},
volume = {50},
number = {7-8},
pages = {469--478},
month = aug,
doi = {10.1016/j.brat.2012.04.007},
pmid = {22659156}
}`)

JSON.stringify(items[0], null, 2)
```

**Expected behavior:**

```json
{
  "id": "test",
  "type": "article",
  "author": [
    {
      "family": "Arch",
      "given": "Joanna J"
    },
    {
      "family": "Wolitzky-Taylor",
      "given": "Kate B"
    },
    {
      "family": "Eifert",
      "given": "Georg H"
    },
    {
      "family": "Craske",
      "given": "Michelle G"
    }
  ],
  "title": "Longitudinal treatment mediation of traditional cognitive behavioral therapy and acceptance and commitment therapy for anxiety disorders.",
  "container-title": "Behaviour research and therapy",
  "volume": "50",
  "issue": "7-8",
  "page": "469–478",
  "DOI": "10.1016/j.brat.2012.04.007",
  "PMID": "22659156",
  "issued": {
    "date-parts": [
      [
        "2012",
        "08",
        ""
      ]
    ]
  }
}
```

**Actual behavior:**

```json
{
  "id": "test",
  "type": "article",
  "author": [
    {
      "family": "Arch",
      "given": "Joanna J"
    },
    {
      "family": "Wolitzky-Taylor",
      "given": "Kate B"
    },
    {
      "family": "Eifert",
      "given": "Georg H"
    },
    {
      "family": "Craske",
      "given": "Michelle G"
    }
  ],
  "title": "Longitudinal treatment mediation of traditional cognitive behavioral therapy and acceptance and commitment therapy for anxiety disorders.",
  "container-title": "Behaviour research and therapy",
  "volume": "50",
  "number": "7-8",
  "page": "469–478",
  "DOI": "10.1016/j.brat.2012.04.007",
  "PMID": "22659156",
  "issued": {
    "date-parts": [
      [
        "2012",
        "08",
        ""
      ]
    ]
  }
}
```

**Comment by hubgit** [comment]
There's also an `issue` field supported by biblatex:

> The issue of a journal. This field is intended for journals whose individual issues
are identified by a designation such as ‘Spring’ or ‘Summer’ rather than the month
or a number. The placement of issue is similar to month and number. Integer
ranges and short designators are better written to the number field. See also month,
number and §§ 2.3.10 and 2.3.11.

http://mirror.ox.ac.uk/sites/ctan.org/macros/latex/contrib/biblatex/doc/biblatex.pdf

It might make sense to map BibTeX's`number` to CSL's `issue` if there's _also_ an `issue` property in the BibTeX that's already being mapped to CSL's `issue`.

**Comment by dsifford** [comment]
Thoughts @retorquere?

**Comment by retorquere** [comment]
> It might make sense to map BibTeX's`number` to CSL's `number` if there's _also_ an `issue` property in the BibTeX that's already being mapped to CSL's `issue`.

That seems sensible to me, but I know relatively little about CSL TBH.

(wrt interpreting the meaning of bib(la)tex fields I'd sooner follow the biblatex manual than the openoffice docs, but in this case they seem in agreement)

**Comment by dsifford** [comment]
Cool, sounds good to me. @hubgit feel free to open a PR. I'd be happy to merge assuming it doesn't cause any unexpected breaks in the tests.

**Comment by retorquere** [comment]
BTW, and I know this is going to sound like a broken record by now, that title field should have been

```
title = {Longitudinal Treatment Mediation of Traditional Cognitive Behavioral Therapy and Acceptance and Commitment Therapy for Anxiety Disorders.},
```

The reference as-is would not render properly under title-case styles like Turabian.

**Comment by hubgit** [comment]
For reference, this is an example of a publisher that uses `number` in BibTeX to contain the issue, and the Crossref API does the same:

* https://link.springer.com/article/10.1007%2Fs12016-016-8572-5
* https://citation-needed.springer.com/v2/references/10.1007/s12016-016-8572-5?format=bibtex&flavour=citation
* https://api.crossref.org/works/10.1007/s12016-016-8572-5/transform/application/x-bibtex

**Comment by retorquere** [comment]
A lot of publishers treat bibtex as an afterthought and have absolutely no experience in outputting correct (and occasionally, not even valid) bib(la)tex.

**Comment by hubgit** [comment]
> A lot of publishers treat bibtex as an afterthought and have absolutely no experience in outputting correct (and occasionally, not even valid) bib(la)tex.

True, but in this case it is actually correct. I was really pointing out that publishers do actually use the core BibTeX fields and not the extended fields (including `issue`) that are supported by biblatex.

**Comment by retorquere** [comment]
That is correct -- a lot of publishers are also solidly stuck in the past here. In my opinion, biblatex should have supplanted bibtex aeons ago 😉 . But it is true that many publishers still demand bibtex (and in their defense, there can be good reasons for this).

In my view, biblatex is not an extension on bibtex, it's its own thing. biblatex does have a soft goal of being able to use bibtex input but doesn't have a hard design goal to be fully backwards compatible with bibtex. I'm not even sure biblatex *could* decide to be 100% backwards compatible with bibtex, because bibtex is underspecified, and on top of that a lot of its behavior sits in style files that biblatex doesn't use.

In the case of astrocite, I don't see a structural choice on whether it intends to parse primarily bibtex or biblatex, nor do I think it should necessarily.

**Comment by hubgit** [comment]
Closing as this was implemented in #29
